drivers/store/memory/cache.go

```diff
@@ -50,34 +51,61 @@ func startCleaner(cache *Cache, interval time.Duration) {
 	go cleaner.Run(cache)
 }
 
-// Counter is a simple counter with an optional expiration.
+// Counter is a simple counter with an expiration.
 type Counter struct {
-	Value      int64
-	Expiration int64
+	mutex      sync.RWMutex
+	value      int64
+	expiration int64
 }
 
 // Expired returns true if the counter has expired.
-func (counter Counter) Expired() bool {
-	if counter.Expiration == 0 {
-		return false
+func (counter *Counter) Expired() bool {
+	counter.mutex.RLock()
+	defer counter.mutex.RUnlock()
+
+	return counter.expiration == 0 || time.Now().UnixNano() > counter.expiration
+}
+
+// Load returns the value and the expiration of this counter.
+// If the counter is expired, it will use the given expiration.
+func (counter *Counter) Load(expiration int64) (int64, int64) {
+	counter.mutex.RLock()
+	defer counter.mutex.RUnlock()
+
+	if counter.expiration == 0 || time.Now().UnixNano() > counter.expiration {
```

Comment:
why not re-use directly `Expired()` here?

Reply:
Because I need to return `counter.value` and `counter.expiration` under a read mutex. I could have created a private function called `expired()` that does exactly that, but it was simpler in my opinion to copy / paste that than to create a method. No strong opinion here, I can change that.